lib/Parse/ParseStmt.cpp

```diff
@@ -348,6 +374,23 @@ ParserStatus Parser::parseBraceItems(SmallVectorImpl<ASTNode> &Entries,
       diagnose(EndOfPreviousLoc, diag::statement_same_line_without_semi)
         .fixItInsert(EndOfPreviousLoc, ";");
       // FIXME: Add semicolon to the AST?
+      Token previousToken = Lexer::getTokenAtLocation(SourceMgr, PreviousLoc);
+      Optional<tok> stmtTokMatch = None;
+      if (previousToken.getKind() == tok::identifier) {
+        SourceLoc startOfLineLocation = L->getLocForStartOfLine(SourceMgr, previousToken.getLoc());
+        SourceLoc locationOfTokenAtStartOfLine = Lexer::getTokenAtLocation(SourceMgr, startOfLineLocation).getLoc();
+        if (previousToken.getLoc() == locationOfTokenAtStartOfLine) {
+          stmtTokMatch = closestMatchingStmtTokenForIdentifier(previousToken.getText());
+        }
+      }
+      if (stmtTokMatch && Tok.isAny(tok::kw_let, tok::kw_var)) {
+        StringRef stmtTokenText = getTokenText(*stmtTokMatch);
+        diagnose(PreviousLoc, diag::statement_misspell, stmtTokenText)
+          .fixItReplace(PreviousLoc, stmtTokenText);
+        consumeToken();
+        skipUntilDeclStmtRBrace(tok::r_brace);
```

Comment:
Ah, I meant something like:
```cpp
if (stmtTokMatch) {
  ... diagnose ...
  if (Tok.isAny(tok::kw_let, tok::kw_var))
    consumeToken();
  skipUntilDeclStmtRBrace(tok::r_brance);
}
```
We still can recover non `let`/`var` binding cases.

Reply:
@rintaro Sure. Fixed.